services/databaseEncryptionService.ts

```diff
@@ -23,55 +22,73 @@ export class EncryptionError extends Error {
 export const ERROR_CODES = {
   USER_CANCELLED: 'USER_CANCELLED',
   KEY_NOT_FOUND: 'KEY_NOT_FOUND',
+  UNINITIALIZED_ENCRYPTION: 'UNINITIALIZED_ENCRYPTION',
 } as const;
 
-let keyCache: Uint8Array | null = null;
+let keyCache: string | null = null;
 let initializationPromise: Promise<void> | null = null;
+let reWrapPromise: Promise<void> | null = null;
 
-function generateRandomKey(): Uint8Array {
-  return Crypto.getRandomBytes(32);
+function generateRandomKeyHex(): string {
+  const bytes = Crypto.getRandomBytes(32);
+  return Array.from(bytes).map(b => b.toString(16).padStart(2, '0')).join('');
 }
 
+// expo-secure-store v15.0.7 throws ERR_AUTHENTICATION for both:
+// - Authentication required (key needs auth but requireAuthentication=false was used)
+// - User cancellation (user cancelled the biometric prompt)
+// Tested on Android. There's no reliable way to distinguish between these two scenarios.
+// We use the same check but keep separate functions for semantic clarity.
 function isCancellationError(error: unknown): boolean {
+  const code = (error as any)?.code;
+  return code === 'ERR_AUTHENTICATION';
+}
```

Comment:
This function makes less sense to me as it changes. Not it's not even doing what it says on the tin -- it's not even to do with cancellation anymore.